app/models/video.rb

```diff
@@ -10,4 +14,67 @@ def self.search_by_title(search = "")
     search.blank? ? all : where('title iLIKE ?', "%#{search}%").order(created_at: :asc)
   end
 
+  def average_rating
+    reviews.average(:rating).to_f.round(1) if reviews.any?
+  end
+  
+  def self.search(query, options={})
+    search_definition = base_query(query)
+    add_field_to_query(search_definition, 'reviews.content') if options[:reviews].present?
+    add_rating_filter(search_definition, options) if rating_included?(options)
```

Comment:
Moving the logic to private methods is definitely a good approach.

The next step would be to have a separate dedicated class to build search definitions.